### PR TITLE
Migrate from Travis CI to GitHub Actions CI

### PR DESCRIPTION
**Pull request by DilumAluthge**
1. Migrate to GitHub Actions CI
2. Disable Travis CI
3. Disable AppVeyor CI
2. Increase code coverage to 93%
3. Update the TagBot configuration

### CODE REVIEW

**Comment by DilumAluthge** [comment]
Alright, this is ready to review.

**Comment by DilumAluthge** [comment]
cc: @eschnett

**Comment by eschnett** [comment]
I was tricked by the Github review UI. I wrote my comments yesterday, but didn't submit them.

File: appveyor.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 environment:
```
[the remark this answers is not in the document]
Comment (DilumAluthge) [reply appveyor.yml:1]: Sure! I'll delete the `appveyor.yml` file.

You'll also need to go to the AppVeyor website and disable AppVeyor on this repository.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+jobs:
+  test:
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} - ${{ github.event_name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        arch:
+          - x64
+          - x86
```
[the remark this answers is not in the document]
Comment (DilumAluthge) [reply .github/workflows/ci.yml:19]: Unfortunately, there are not :( 

At least not right now.

**Comment by DilumAluthge** [comment]
I've deleted the `appveyor.yml` file. I think you need to delete the AppVeyor webhooks from the repo.

If you go to https://github.com/eschnett/SIMD.jl/settings/hooks, you'll see all the webhooks. You can delete all the AppVeyor webhooks.

File: appveyor.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (DilumAluthge) [reply appveyor.yml]: I've deleted the `appveyor.yml` file.

**Comment by eschnett** [comment]
Thanks a lot!

File: .github/workflows/ci_julia_nightly.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
+name: CI (Julia nightly)
```
Comment (KristofferC) [inline .github/workflows/ci_julia_nightly.yml:1]: Why does this need to be a separate file?

Comment (DilumAluthge) [reply .github/workflows/ci_julia_nightly.yml:1]: It's the only way to get a status badge that is green even if tests fail on Julia nightly.

If you don't mind a status badge that turns red if tests fail on Julia nightly but pass on all other Julia versions, then we can merge those two workflow files into a single workflow file.

Comment (KristofferC) [reply .github/workflows/ci_julia_nightly.yml:1]: Alright, thanks. Might as well keep it like this then.

**Pull request merged**